--- a/src/xwidget.c
+++ b/src/xwidget.c
@@ -362,8 +362,12 @@
 				"download-started",
 				G_CALLBACK (webkit_download_cb), xw);

+#if !defined HAVE_PGTK
+              /* when using pgtk, the about:blank workaround is not needed
+                 would in fact make the initial load fail.  */
 	      webkit_web_view_load_uri (WEBKIT_WEB_VIEW (xw->widget_osr),
 					"about:blank");
+#endif
 	      /* webkitgtk uses GSubprocess which sets sigaction causing
 		 Emacs to not catch SIGCHLD with its usual handle setup in
 		 'catch_child_signal'.  This resets the SIGCHLD sigaction.  */
